test/testcafe/spec/SessionStorage_spec.js

```diff
@@ -209,3 +212,76 @@ test.requestHooks(identifyChallengeMock)('shall clear when session.stateHandle i
   await t.expect(identityPage.form.getTitle()).eql('Sign In');
   await t.expect(getStateHandleFromSessionStorage()).eql(null);
 });
+
+
+test.requestHooks(identifyChallengeMock)('will clear when session.stateHandle has different app context', async t => {
+  let useNewTokenResponse = false;
+  const multipleIntrospectMock = RequestMock()
+    .onRequestTo('http://localhost:3000/idp/idx/introspect')
+    .respond((req, res) => {
+      if (useNewTokenResponse) {
+        // mimic response for new token
+        res.statusCode = '200';
+        res.setBody(xhrIdentify);
+      } else {
+        // return a different app context
+        res.statusCode = '200';
+        const xhrIdentifyWithDifferentApp =  {
+          ...xhrIdentify,
+          app: {
+            value: {
+              id: '123',
+            }
+          },
+          stateHandle: 'stateHandleForAppId123',
+        };
+        res.setBody(xhrIdentifyWithDifferentApp);
+        useNewTokenResponse = true;
+      }
+    });
+  const identityPage = new IdentityPageObject(t);
+  const challengeEmailPageObject = new ChallengeEmailPageObject(t);
+
+  // Identify page
+  await identityPage.navigateToPage();
+  await t.expect(getStateHandleFromSessionStorage()).eql(null);
+  await identityPage.fillIdentifierField('foo@test.com');
+  await identityPage.clickNextButton();
+
+  // Email challenge page
+  const pageTitle = challengeEmailPageObject.form.getTitle();
+  await t.expect(pageTitle).eql('Verify with your email');
+  await t.expect(getStateHandleFromSessionStorage()).eql(xhrEmailVerification.stateHandle);
+
+  // Reset mocks
+  await t.removeRequestHooks(identifyChallengeMock);
+  await t.addRequestHooks(multipleIntrospectMock);
+  await t.addRequestHooks(introspectRequestLogger);
+
+  // Refresh
+  await challengeEmailPageObject.refresh();
+
+  // Verify introspect requests
+  // introspect with session.stateHandle
```

Comment:
mention that in this flow case both `/introspect` are valid and success.. because stateHandle is still valid